### PR TITLE
Restructured to monorepo

### PR DESCRIPTION
**Pull request by spoenemann**
This restructures the directory structure of this repository to a monorepo (see #243). Once the PR is accepted, I will add more packages to the repository.

### CODE REVIEW

**Review by JanKoehnlein — APPROVED** [review]
LGTM

File: lerna.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  "useWorkspaces": true,
+  "npmClient": "yarn",
+  "command": {
+    "run": {
+      "stream": true
+    }
+  },
+  "packages": [
+    "packages/*"
+  ]
```
Comment (planger) [inline lerna.json:12]: I think this is redundant, as we specify `useWorkspaces`.

File: package.json [hunk 2]
```diff
@@ -1,123 +1,16 @@
...
-    "clean": "rimraf lib artifacts",
-    "build": "tsc -p ./tsconfig.json && yarn run lint",
-    "lint": "eslint -c configs/.eslintrc.js \"src/**/!(*.spec.ts*|test-helper.ts)\"",
-    "test:cli": "ts-mocha \"./src/**/*.spec.?(ts|tsx)\"",
-    "watch": "tsc -w -p ./tsconfig.json",
-    "test": "jenkins-mocha --opts ./configs/mocha.opts \"./src/**/*.spec.?(ts|tsx)\"",
-    "prepare": "yarn run clean && yarn run build",
-    "examples:build": "webpack --config ./configs/webpack.config.js",
-    "examples:watch": "webpack --watch --config ./configs/webpack.config.js",
-    "examples:run": "http-server examples",
```
Comment (planger) [inline package.json:104]: It might be useful to still keep those scripts. It may help newcomers to identify more easily what to try.

**Review by planger — APPROVED** [review]
Thanks, looks excellent! Just a few minor suggestions inline, but they aren't critical.

**Review by tortmayr — APPROVED** [review]
LGTM! :+1:

**Review by tomvdbussche — APPROVED** [review]
LGTM!

File: package.json (hunk 2 above)
Comment (spoenemann) [reply package.json:104]: They are now in a separate package in the `examples` folder.

Comment (spoenemann) [reply package.json:104]: We should document that in the README.

**Pull request merged**